### DISCUSSION TITLE
ISO data licensing issues

### DISCUSSION THREAD

**Discussion by bendnorman** [catalyst-cooperative/pudl#1750] [Ideas]
This topic is for discussing unclear licensing of data released by Independent System Operators (ISO). 

Users often ask us if ISO data is available through PUDL. We've looked into adding the troves of data ISO make "public" but find their licensing isn't clear or permissive. It seems like many organizations scrape ISO data for internal use but don't properly license the data. We aren't comfortable redistributing ISO data without a common permissive license like [CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/).

This is an edited exerpt from a previous PUDL grant about ISO data licensing:

All data from FERC, [EIA](https://www.eia.gov/about/copyrights_reuse.php), [EPA](https://edg.epa.gov/EPA_Data_License.html), and other US Government agencies are in the public domain, and our derivative works based solely upon US Government data are released under a Creative Commons Attribution 4.0 ([CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/)) license or placed in the public domain with an explicit Creative Commons Public Domain Dedication.

## Background on ISO licensing
Catalyst is interested in publishing the ISO/RTO LMP data under the most permissive licensing possible, however, the input data to be sourced from the regional grid operators is subject to a wide variety of licensing terms and conditions. The California Independent System Operator ([CAISO](http://www.caiso.com/Pages/PrivacyPolicy.aspx)), Energy Reliability Council of Texas ([ERCOT](http://www.ercot.com/help/terms)), and Southwest Power Pool ([SPP](https://www.spp.org/terms-conditions/)) all make explicit provisions for sharing and re-use of their data, at least in the case of non-commercial use. The Northeast ISO ([NEISO](https://www.iso-ne.com/legal-privacy)), New York ISO ([NYISO](https://www.nyiso.com/legal-notice)), The PJM Interconnection, and the Midcontinent ISO ([MISO](https://www.misoenergy.org/about/legal-and-privacy/)) appear to have much more restrictive licensing, prohibiting any re-use or dissemination of their data. However, other academic researchers, open-source projects, and commercial services are already clearly making use of this data for a variety of purposes (e.g. [WattTime’s PyISO library](https://github.com/WattTime/pyiso), [Tomorrow’s real-time Electricity Map](https://github.com/tmrowco/electricitymap-contrib) and Zach Tzavelis’s @m4rz910 [NYISOToolkit](https://github.com/m4rz910/NYISOToolkit)). For more background on the issue of restrictive or vaguely worded licensing requirements for quasi-public energy data in a research context, please see the legal usability section of [Open Data for Energy Modeling](https://www.bmwk.de/Redaktion/EN/Publikationen/Studien/open-Data-for-electricity-modeling.html) and [this topic on the OpenMod Forum](https://forum.openmod.org/t/unclear-ambiguous-and-or-contradictory-licensing-by-public-bodies/3102).

## Licensing Language from each ISO

Original Data Source Licensing:

- **FERC:** US Govt. Work / Public Domain.

- **US EIA**: US Govt. Work / Public Domain.

- **US EPA:** US Govt. Work / Public Domain.

- **[CAISO:](http://www.caiso.com/Pages/PrivacyPolicy.aspx)** _“Most of the materials and information contained on this Website and the California ISO mobile application were generated, compiled or assembled from materials and information that are freely available for public use consistent with the general policies of the Public Records Act (California Government Code section 6250, et seq.), as provided by California Public Utilities Code section 345.5, and may be used by you provided that you keep intact all copyright, trademark and other proprietary notices and that you credit the California ISO when using such materials and/or information.”_

- **[ERCOT:](https://www.ercot.com/help/terms)** _“The information presented on this website has been compiled by or for ERCOT for general information purposes only. The publicly available contents of this website may be used, reproduced and redistributed, provided that the contents are not modified and that you maintain all copyright and other notices contained in the contents, including this Agreement. Notwithstanding the foregoing, raw data provided in public portions of this website may be used, reproduced and redistributed in compilations, charts and analyses without maintaining such notices. ERCOT does not guarantee the accuracy of any such compilation, chart or analysis.”_

- **[MISO:](https://www.misoenergy.org/about/legal-and-privacy/)** _“You may not modify, publish, transmit, participate in the transfer or sale of, reproduce, create derivative works of, distribute, publicly perform, publicly display or in any way exploit any of the materials or content or MISO's sites in whole or in part. Your access to MISO's sites does not confer any license or ownership interest in either the form or content of MISO's sites, including any confidential information, proprietary information, or intellectual property of any kind or nature; MISO hereby expressly reserves such rights and property in their entirety.”_

- **[NEISO:](https://www.iso-ne.com/legal-privacy)** _“You are also hereby put on notice that the Content is protected by copyright under United States laws. Any duplication of the Content or non-personal use may violate copyright, trademark, and other laws.”_

- **[NYISO:](https://www.nyiso.com/legal-notice)** _“Access to this Web site does not confer any license or ownership interest in either the form or content of the Web site, including any confidential or proprietary information or intellectual property of any kind or nature, and the NYISO hereby expressly reserves such rights and property in its entirety.”_

- **PJM: (Source link is dead, needs update)** _“The Materials and Services are protected by copyright and/or other intellectual property laws and any unauthorized use of the Materials or Services may violate such laws. Except as expressly provided herein, PJM Connext does not grant any express or implied right or license or ownership of any kind to you under any patents, copyrights, trademarks, or trade secret information with respect to the Materials and Services. Except as expressly provided by copyright law or these Terms of Use, you may not copy, distribute, modify, publish, sell, transfer, license, transmit, display, participate in the transfer or sale of, create derivative works of, or in any way exploit (commercially or otherwise) any of the Materials or Services, either in whole or in part. You acknowledge that you do not acquire any ownership rights by downloading copyrighted material.” PJM-issued Redistribution License required for many applications._ 

- **[SPP:](https://www.spp.org/terms-conditions/)** _“All materials posted or otherwise available on this website are the exclusive copyrighted material of the author(s) or Southwest Power Pool, Inc. (SPP). Permission is implicitly granted to copy and distribute (via computer network or printed form) in whole or in part (with appropriate citation) EXCEPT when such materials will be used, in whole or in part, within a commercial publication (printed or otherwise) or when the author(s) or SPP will be quoted in commercial materials, forums or publications. Any commercial use of these materials requires prior, express written authorization from the author(s) or a duly authorized officer of SPP.”_

## Questions
We'd love to integrate ISO data but don't feel comfortable republishing the data given the licensing issues listed above. Have other folks talked with the ISOs about these issues? Is it possible to require ISOs to apply permissive licensing for the data because they are quasi governmental organizations? Let us know what you think!

**Comment by zaneselvans** [comment]
I think software which can access the various ISO/RTO APIs is probably less illegal, e.g. WattTime's PyISO project -- but if that data is stored and republished, especially for commercial purposes, that's probably much less okay. Especially if that proprietary data gets mixed in with other open data you can have problems with tainted data products downstream -- where everything derived from (anything derived from) anything derived from ... proprietary data is also illegal.

There's also a weird subtlety with the US Govt. Works licenses, in that it only explicitly applies within the US.  The US Govt. **can** assert copyright over its works in other jurisdictions.  But in practice this data is generally treated as public domain.

**Reply by m4rz910** [reply]
I wasn't aware of the possible licensing issue when I made NYISOToolkit available on Github. It sounds like I should probably take it down until there's more clarity on licensing? On the other hand, there are organizations like YesEnergy and ABB that provide data services with access to some of this data - so clearly they're distributing this data commercially. Question is whether they have signed agreements with the ISOs to do this?

**Reply by zaneselvans** [reply]
@patrickbrown4 (then at MIT, now at NREL) also compiled 7 years of hourly ISO pricing data and [archived it on Zenodo](https://zenodo.org/record/3562896), and painstakingly extracted permission from all the ISOs (except PJM) to do so for "research purposes" (I think) in conjunction with [this 2020 paper](https://ceepr.mit.edu/workingpaper/spatial-and-temporal-variation-in-the-value-of-solar-power-across-united-states-electricity-markets/) so he could probably speak to what that process was like

[His Google Scholar](https://scholar.google.com/citations?hl=en&user=92he7yAAAAAJ&view_op=list_works&sortby=pubdate)

**Reply by zaneselvans** [reply]
@m4rz910 I have to imagine that big corporate resellers like S&P or ABB have signed licensing / resale deals with the ISOs. I think code that accesses the APIs should be completely fine, it's just the data itself that's sketchy.

**Reply by patrickbrown4** [reply]
I'm definitely no expert on the legal side and am not actively working with this kind of data at the moment, but if you ping me at prbrown [at] mit [dot] edu I can forward you some of my exchanges with the ISOs as a starting point.

**Reply by zaneselvans** [reply]
Thanks Patrick! I'll send you a ping and CC @bendnorman

**Reply by MichaelTiemann** [reply]
Out of curiosity I looked up MISO, who publish their financial reports.  In their 2021 report it's not clear that data licensing is an actual material component of revenues or assets.  Since it is a member organization, perhaps members can recommend a change to the license that favors public disclosure and research.  The 2021 annual report is here: https://cdn.misoenergy.org/2021%20MISO%20Year%20End%20Financial%20Statements%20624392.pdf

**Reply by m4rz910** [reply]
FYI Iv found some more free tools out there recently...

https://github.com/kmax12/isodata

https://github.com/WattTime/pyiso

**Reply by zaneselvans** [reply]
@kmax12 you might be interested in this discussion on ISO/RTO licensing and some of the linked articles / discussions.
